server/src/main/java/org/elasticsearch/search/aggregations/metrics/InternalExtendedStats.java

```diff
@@ -131,14 +147,29 @@ public double getVariance() {
 
     @Override
     public double getVariancePopulation() {
-        double variance =  (sumOfSqrs - ((sum * sum) / count)) / count;
-        return variance < 0  ? 0 : variance;
+        if(count < 1){
+            return Double.NaN;
+        }
+        if(Double.isNaN(m2)){
```

Comment:
I think it is worth a comment that this can happen if we combine with a node that doesn't calculate m2.